lib/game.rb

```diff
@@ -0,0 +1,38 @@
+class Game
+  def initialize(rolls)
+    @rolls = rolls
+  end
+
+  def score
+    score = 0
+    @rolls.each_with_index do |frame, i|
+      score += frame_score(frame, i)
+    end
+    score
+  end
+
+  private
+
+  def frame_score(frame, index)
+    return 0 if index > 9                                    # don't count the bonus on their own
+    frame_score = frame.inject(0,:+)
+    frame_score += strike_or_spare(frame, index) if frame_score == 10 # if a strike or spare
+    frame_score
+  end
+
+  def strike_or_spare(frame, index)
+    if frame.length == 1                                    # if a strike
+      strike_score(index)
+    else                                                    # if a spare
```

Comment:
You could potentially protect this method being used in future versions of the code by including a guard against frames scoring less than 10, where they return 0